### PR TITLE
Add "See Less" functionality

### PR DESCRIPTION
**Pull request by nlfurniss**
Adds the ability to truncate the text via a "see less" button and two new optional user defined actions: `onCollapse` and `onToggleTruncate`.

### CODE REVIEW

File: addon/templates/components/truncate-multiline.hbs [hunk 1]
```diff
@@ -1,18 +1,21 @@
 {{#if truncate}}
   <div class="truncate-multiline--truncation-target">
     {{~#if hasBlock~}}
-      {{yield}}
+      {{~yield~}}
```
Comment (nickiaconis) [inline addon/templates/components/truncate-multiline.hbs]: These tildes are unnecessary b/c of the surrounding tildes. Tilde removes whitespace up to the next non-whitespace thing, which means back-to-back tildes are redundant.

File: tests/integration/components/truncate-multiline-test.js [hunk 2]
```diff
@@ -80,25 +75,30 @@ moduleForComponent('truncate-multiline', 'Integration | Component | truncate-mul
...
     <div style="width: 362px; font: 16px sans-serif;">
-      {{truncate-multiline buttonText="click me" text="supercalifragilisticexpialidocious supercalifragilisticexpialidocious supercalifragilisticexpialidocious supercalifragilisticexpialidocious"}}
+      {{truncate-multiline seeMoreButtonText="click me" seeLessButtonText="then click me" text="supercalifragilisticexpialidocious supercalifragilisticexpialidocious supercalifragilisticexpialidocious supercalifragilisticexpialidocious"}}
     </div>
   `);
 
+  let this$ = this.$().clone();
+  // remove attributes we have no control over
+  this$.find('[id^=ember]').removeAttr('id');
+  this$.find('[data-ember-action]').removeAttr('data-ember-action');
```
Comment (nickiaconis) [inline tests/integration/components/truncate-multiline-test.js]: I would suggest using `$this`. `this$` is very close to `this.$` visually.

File: tests/integration/components/truncate-multiline-test.js [hunk 3]
```diff
@@ -208,5 +230,32 @@ moduleForComponent('truncate-multiline', 'Integration | Component | truncate-mul
...
+    </div>
+  `);
+
+  return wait().then(() => {
+    this.$('button').click();
+
+    return wait().then(() => {
+      this.$('button').click();
+    });
+  });
```
Comment (nickiaconis) [inline tests/integration/components/truncate-multiline-test.js]: Promise chaining might make this easier to read:

```javascript
return wait().then(() => {
  ...
  return wait();
}).then(() => {
  ...
});
```

File: addon/templates/components/truncate-multiline.hbs (hunk 1 above)
Comment (nlfurniss) [inline addon/templates/components/truncate-multiline.hbs]: Looks like I didn't do a `pull` before submitting this :-(

Will fix

File: tests/integration/components/truncate-multiline-test.js (hunk 3 above)
Comment (nlfurniss) [inline tests/integration/components/truncate-multiline-test.js]: Makes sense

**Comment by nlfurniss** [comment]
updated based on owner feedback

**Comment by nickiaconis** [comment]
Thanks for the updates. I've fixed the Travis builds. Can you please rebase?

**Comment by nlfurniss** [comment]
Updated to make `_showSeeLessButton` a cp.

File: package.json [hunk 4]
```diff
@@ -52,4 +52,4 @@
   "ember-addon": {
     "configPath": "tests/dummy/config"
   }
-}
+}
```
Comment (nickiaconis) [inline package.json]: Can you drop this change please?

**Comment by nickiaconis** [comment]
LGTM (aside from extraneous package.json change)

**Pull request merged**